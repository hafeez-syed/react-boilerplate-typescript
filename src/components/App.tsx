import * as React from 'react';
import { hot } from 'react-hot-loader';
import PTypes from 'prop-types';
import axios from 'axios';

import Header from './layout/Header';
import Main from './Main';
import Footer from './layout/Footer';

interface IState {
  count: number;
  asyncCounters: number;
}

const title = 'React, Webpack, Babel, Dev-server Boilerplate !!!...';

export const doIncrement = prevState => ({
  count: prevState.count + 1
});

export const doDecrement = prevState => ({
  count: prevState.count - 1
});

class App extends React.Component<{}, IState> {
  unmounted = false;
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
      asyncCounters: 0
    };
  }

  componentDidMount() {
    axios
      .get('https://reqres.in/api/users?page=2', {
        headers: {
          'Access-Control-Allow-Origin': '*'
        }
      })
      .then(counter => {
        if (this.unmounted) return;
        this.setState({ asyncCounters: counter.data.total });
      });
  }

  componentWillUnmount() {
    this.unmounted = true;
  }

  increment = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    event.preventDefault();
    this.setState(doIncrement);
  };

  decrement = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    event.preventDefault();
    this.setState(doDecrement);
  };

  render() {
    const { count } = this.state;

    return (
      <React.Fragment>
        <Header />
        {/* <Main
          title={title}
          count={count}
          increment={this.increment}
          decrement={this.decrement}
        >
          <AsyncCounters asyncCounters={this.state.asyncCounters} />
        </Main> */}
        <Footer />
      </React.Fragment>
    );
  }
}

export const AsyncCounters = ({ asyncCounters }) => (
  <label>Total data: {asyncCounters}</label>
);

AsyncCounters.propTypes = {
  asyncCounters: PTypes.number.isRequired
};

export default hot(module)(App);
