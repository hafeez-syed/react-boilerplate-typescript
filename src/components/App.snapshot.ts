import * as React from 'react'
import renderer from 'react-test-renderer'

import App, { AsyncCounters } from './App'

describe('App Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(<App />)
    component.getInstance().setState({ asyncCounters: 1 })

    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('Counter Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(<AsyncCounters asyncCounters={1} />)

    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
