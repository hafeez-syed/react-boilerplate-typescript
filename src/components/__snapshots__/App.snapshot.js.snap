// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot renders 1`] = `
Array [
  <header
    role="banner"
  >
    <p>
      Hello Company-Logo
    </p>
  </header>,
  <nav
    role="navigation"
  >
    <ul>
      <li>
        Put navigation here
      </li>
    </ul>
  </nav>,
  <main
    role="main"
  >
    <h1>
      React, Webpack, Babel, Dev-server Boilerplate !!!...
    </h1>
    <h2
      className={null}
    >
      0
    </h2>
    <button
      onClick={[Function]}
    >
      Add ++
    </button>
    <button
      onClick={[Function]}
    >
      Sub --
    </button>
  </main>,
  <footer
    role="contentinfo"
  >
    <p>
      Put copyright, etc. here.
    </p>
  </footer>,
]
`;

exports[`Counter Snapshot renders 1`] = `
<label>
  Total data: 
  1
</label>
`;
